ngraph/test/util/ndarray.hpp

```diff
@@ -223,5 +225,17 @@ namespace ngraph
                     NDArrayBase<T>::m_elements.begin(), NDArrayBase<T>::m_shape, initial_value);
             }
         };
+
+        /// An N dimensional array of elements of type T
+        template <typename T, size_t N>
+        NDArrayBase<T> nd_array(const init::NestedInitializerList<T, N>& initial_value)
+        {
+            NDArrayBase<T> array(init::get_shape<T, N>(initial_value));
+
+            init::flatten<typename std::vector<T>::iterator, T, N>(
+                array.begin(), array.get_shape(), initial_value);
+
+            return array;
+        }
```

Comment:
@ilyachur , @tomdol ,
I want to add this helper function for array creation. It allow to parametrize unit test more easily. Is it OK for you?

Reply:
Why cannot you use the constructor of `NDArray`?

Reply:
```
        template <typename T, size_t N>
        class NDArray : public NDArrayBase<T>
```
NDArray use `size_t N` which gives different array type for different sizes. I want to use `NDArrayBase<T>` as a parameter and drop size info from type. `NDArrayBase<T>` has shape description and this is enough to drive the tests.

This approach allow to create parametrize test where size is use in runtime and doesn't force to have templatize test parameters.

In this PR
```
    template <typename ValueType>
    struct PadBackendTestParams
```
will change if we use NDArray to 
```
    template <typename ValueType, size_t InDim, size_t OutDim, size_t PadDim>
    struct PadBackendTestParams
```

Reply:
@ilyachur , is my explanation satisfy you question?

Reply:
No I don't fully understand the reason. As I see at the current moment you are using `nd_array()` method in order to create a Parameter:
```cpp
                               testing::Values(Params<float>{test::nd_array<float, 2>({
                                                                 {1, 2, 3},
                                                                 {4, 5, 6},
                                                                 {7, 8, 9},
                                                             }),
                                                             test::nd_array<int64_t, 1>({1, 2}),
                                                             test::nd_array<int64_t, 1>({2, 1}),
                                                             test::nd_array<float, 2>({
                                                                 {2, 1, 1, 2, 3, 3},
                                                                 {2, 1, 1, 2, 3, 3},
                                                                 {5, 4, 4, 5, 6, 6},
                                                                 {8, 7, 7, 8, 9, 9},
                                                                 {8, 7, 7, 8, 9, 9},
                                                                 {5, 4, 4, 5, 6, 6},
                                                             }),
                                                             op::PadMode::SYMMETRIC},
```
But if you change `nd_array` to `NDArray` it also should work and `NDArray<T, N>` will be casted to `NDArrayBase<T>.
I don't see the big difference.
I don't think that any changes are required for `Parameter<ValueType>`.

Reply:
Drop `nd_array` function.